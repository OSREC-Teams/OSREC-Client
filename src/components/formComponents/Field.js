import styled from 'styled-components';

import { Field } from 'formik';

export default styled(Field)`
  font-size: 1em;
  margin-bottom: 10px;
  ${props => (props.error ? 'border-color: #FF0000' : '')};
  border-radius: 0.3rem;
  height: 2em;
  padding: 0.5em;
  width: 70%;
  text-align: center;

  &:focus {
    outline: #6c3b91;
  }
`;
