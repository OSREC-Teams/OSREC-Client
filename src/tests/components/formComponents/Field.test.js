import React from 'react';
import Field from 'components/formComponents/Field';
import { shallow } from 'enzyme';

test('Button renders a field with error', () => {
  const field = shallow(<Field type="password" name="password" placeholder="Password" error />);
  expect(field).toMatchSnapshot();
});

test('Button renders a field without error', () => {
  const field = shallow(<Field type="password" name="password" placeholder="Password" error={false} />);
  expect(field).toMatchSnapshot();
});
