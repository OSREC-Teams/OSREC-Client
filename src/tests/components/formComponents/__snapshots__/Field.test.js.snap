// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a field with error 1`] = `
<StyledComponent
  error={true}
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "WrappedComponent": [Function],
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-size: 1em;
  margin-bottom: 10px;
  ",
          [Function],
          ";
  border-radius: 0.3rem;
  height: 2em;
  padding: 0.5em;

  &:focus {
    border-color: #6c3b91;
  }
",
        ],
      },
      "displayName": "Styled(C)",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
  name="password"
  placeholder="Password"
  type="password"
/>
`;

exports[`Button renders a field without error 1`] = `
<StyledComponent
  error={false}
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "WrappedComponent": [Function],
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  font-size: 1em;
  margin-bottom: 10px;
  ",
          [Function],
          ";
  border-radius: 0.3rem;
  height: 2em;
  padding: 0.5em;

  &:focus {
    border-color: #6c3b91;
  }
",
        ],
      },
      "displayName": "Styled(C)",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
  name="password"
  placeholder="Password"
  type="password"
/>
`;
